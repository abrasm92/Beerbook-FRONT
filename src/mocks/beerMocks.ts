import { BeerDataApi } from "../types/interfaces";

export const singleBeer: BeerDataApi = {
  id: "6297a8369ab4b03687f15b72",
  name: "Sofie",
  brewery: "Goose Island",
  style: "Belgian season",
  degrees: 6.5,
  IBU: 25,
  country: "EEUU",
  description:
    "Cerveza de color ámbar, con aromas intensos de cítricos, con final lig…",
  image: "",
  imageBackup: "",
  owner: "admin",
};

export const groupOfBeer: BeerDataApi[] = [
  {
    id: "6297a8369ab4b03687f15b72",
    name: "Sofie",
    brewery: "Goose Island",
    style: "Belgian season",
    degrees: 6.5,
    IBU: 25,
    country: "EEUU",
    description:
      "Cerveza de color ámbar, con aromas intensos de cítricos, con final lig…",
    image: "",
    imageBackup: "",
    owner: "",
  },
  {
    id: "6297a8369ab4b03687f15b73",
    name: "Breakfast Stout",
    brewery: "Founders",
    style: "Imperial Stout",
    degrees: 8.3,
    IBU: 60,
    country: "EEUU",
    description:
      "Cerveza negra con gran espuma color marrón con arómas predominantes a …",
    image: "",
    imageBackup: "",
    owner: "",
  },
  {
    id: "6297a8369ab4b03687f15b7a",
    name: "Ginette Natural Christmas",
    brewery: "Brasserie La Binchoise",
    style: "Belgian Strong Ale",
    degrees: 8.5,
    IBU: NaN,
    country: "Bélgica",
    description:
      "Cerveza rojiza oscura con espuma densa, cuerpo medio suave y aromatica…",
    image: "",
    imageBackup: "",
    owner: "",
  },
];
