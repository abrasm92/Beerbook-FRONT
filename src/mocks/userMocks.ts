import { BasicUser, LoginUser, User, UserData } from "../types/interfaces";

export const singleUser: User = {
  name: "admin",
  username: "admin",
  email: "admin@admin.com",
  password: "beerbook",
};

export const loginUser: LoginUser = {
  username: "admin",
  password: "beerbook",
};

export const userLogged: BasicUser = {
  id: "1234",
  name: "admin",
};

export const userData: UserData = {
  name: "admin",
  username: "admin",
  email: "admin@admin.com",
  image: "",
  imageBackup: "",
  creations: 10,
  favorites: 23,
  age: 29,
  country: "espa√±a",
  admin: true,
};
