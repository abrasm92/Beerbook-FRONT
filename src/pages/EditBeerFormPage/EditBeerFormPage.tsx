import { useEffect } from "react";
import { useParams } from "react-router-dom";
import CreateEditBeerForm from "../../components/CreateEditBeerForm/CreateEditBeerForm";
import { useAppDispatch, useAppSelector } from "../../redux/hooks";
import { getBeerByIdThunk } from "../../redux/thunks/beerThunks";
import NotFoundPage from "../NotFoundPage/NotFoundPage";
import EditBeerFormPageStyles from "./EditBeerFormPageStyles";

const EditBeerFormPage = (): JSX.Element => {
  const dispatch = useAppDispatch();
  const { id } = useParams();
  const { singleBeer } = useAppSelector((state) => state.beer);
  const { loading } = useAppSelector((state) => state.ui);
  useEffect(() => {
    dispatch(getBeerByIdThunk(id));
  }, [dispatch, id]);

  return (
    <>
      {!loading && (
        <EditBeerFormPageStyles>
          {singleBeer.name && <CreateEditBeerForm beer={singleBeer} />}
        </EditBeerFormPageStyles>
      )}
      {!singleBeer.name && !loading && <NotFoundPage />}
    </>
  );
};

export default EditBeerFormPage;
