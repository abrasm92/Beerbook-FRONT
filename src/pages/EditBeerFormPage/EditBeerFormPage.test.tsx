import { configureStore, createSlice } from "@reduxjs/toolkit";
import { render, screen } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import reactTestRenderer from "react-test-renderer";
import { groupOfBeer, singleBeer } from "../../mocks/beerMocks";
import { store } from "../../redux/store/store";
import EditBeerFormPage from "./EditBeerFormPage";

describe("Given a EditBeerFormPage component", () => {
  describe("When it's rendered", () => {
    test("Then it sould show button to editBeer", () => {
      const editForm = reactTestRenderer.create(
        <Provider store={store}>
          <EditBeerFormPage />
        </Provider>
      );

      expect(editForm).toMatchSnapshot();
    });

    test("Then it should call dispatch 1 time to get id by params", async () => {
      const beerMockSlice = createSlice({
        name: "beer",
        initialState: {
          page: 0,
          listOfBeers: groupOfBeer,
          singleBeer: singleBeer,
        },
        reducers: {},
      });
      const uiMockSlice = createSlice({
        name: "ui",
        initialState: {
          loading: false,
        },
        reducers: {},
      });
      const mockStore = configureStore({
        reducer: { beer: beerMockSlice.reducer, ui: uiMockSlice.reducer },
      });
      const mockId = `${singleBeer.id}`;
      jest.mock("react-router-dom", () => ({
        ...jest.requireActual("react-router-dom"),
        useParams: () => ({
          id: mockId,
        }),
      }));
      const textButton = "Editar cerveza";

      await render(
        <BrowserRouter>
          <Provider store={mockStore}>
            <EditBeerFormPage />
          </Provider>
        </BrowserRouter>
      );

      const buttonSendEdit = screen.getByRole("button", { name: textButton });

      expect(buttonSendEdit).toBeInTheDocument();
    });
  });
});
