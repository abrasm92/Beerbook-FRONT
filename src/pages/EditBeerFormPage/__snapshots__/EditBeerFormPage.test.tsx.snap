// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a EditBeerFormPage component When it's rendered Then it sould show button to editBeer 1`] = `
Array [
  <section
    className="sc-dkzDqf fJnXvb"
  />,
  <div
    className="sc-gsnTZi bRxkTF"
  >
    <div
      id="oopss"
    >
      <div
        id="error-text"
      >
        <img
          alt="404"
          src="https://cdn.rawgit.com/ahmedhosna95/upload/1731955f/sad404.svg"
        />
        <p
          className="number"
        >
          404
        </p>
        <p
          className="text"
        >
          PAGINA NO ENCONTRADA
        </p>
        <p
          className="p-a"
        >
          La página que estás buscando no se pudo encontrar
        </p>
      </div>
    </div>
  </div>,
]
`;
